tracing-appender/src/rolling.rs

```diff
@@ -578,6 +578,12 @@ impl Inner {
                             .expect("prefix is always present"),
                     )
                 })
+                .filter(|entry| {
+                    entry
+                        .metadata()
+                        .and_then(|metadata| metadata.created())
+                        .is_ok()
+                })
```

Comment:
Maybe you can do a `filter_map` operation instead and keep creation times along the files. That way you can remove the `expect` added below and simplify things a bit. What do you think?

Reply:
Hmm I don't think I follow. 
If you meant using `filter_map()` instead of `filter` ->

How to eliminate the `expect` in the below (line 604) ? 
```rust
file.metadata()
                .and_then(|metadata| metadata.created())
``` this structure will always be a `Result` regardless of we have only valid files with timestamps or not. So I believe, `expect` is necessary. 

If you meant another thing, then probably I did not get it :)

Reply:
You are collecting files in a `Result<Vec<DirEntry>>`. You also extract the creation date once on this occasion. You could thus collect a `Result<Vec<(DirEntry, SystemTime)>>` using `filter_map`, and reuse the values already extracted =)

Something like:
```suggestion
                .filter_map(|entry| {
                    let created = entry
                        .metadata()
                        .and_then(|metadata| metadata.created())
                        .ok()?;
                    Some((entry, created))
                })
```

Reply:
Wonderful, thanks!